syntax = "proto3";

package message;

service grpcService{
 rpc sendState(SelectedStates) returns (GlobalState) {}
 rpc sendSingleState(SelectedState) returns (EmptyResponse) {}
 rpc randomSample(RequestRandomIndices) returns (ResponseRandomIndices) {}
}

message SelectedState {
  bytes state = 1;
}

message EmptyResponse {
  string message = 1;
}

message SelectedStates {
  bytes state = 1;
  string setting = 2;
  bytes weights = 3;
  string model_name = 4;
  string data = 5;
  int32 n_class = 6;
}

message GlobalState {
  bytes state = 1;
  float loss = 2;
  float accuracy = 3;
  float mae = 4;
  float mse = 5;
  float rse = 6;
  float rmse = 7;
  float f1_score = 8;
}

message RequestRandomIndices {
  int32 num_clients = 1;  // 디바이스 개수
  int32 num_samples = 2;  // 샘플링할 개수
}

message ResponseRandomIndices {
  repeated int32 device_indices = 1;  // 선택된 디바이스 인덱스 리스트
}